assets/src/editor/events/ticket-assignments-manager/hooks/use-counts-manager.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External imports
+ */
+import { useMemo, useCallback } from '@wordpress/element';
+import { reduce, cloneDeep, isEmpty } from 'lodash';
+import { __ } from '@eventespresso/i18n';
+
+// @todo, when the wp.data.useSelect hook is available, we can utilize that in
+// this custom hook and then we'll be able to remove the withSelect hoc usage
+// in the composed `TicketAssignmentsManagerModal` component.
+
+const useCountsManager = (
+	dateEntities,
+	ticketEntities,
+	assignedCounts,
+	assignedState,
+) => {
+	const getMessage = useCallback( ( whichAreEmpty ) => {
```

Comment:
Instead of `whichAreEmpty` being a ref (which was fragile), it is now passed in as an argument to this memoized callback.  Thus it's much less fragile.